zhaquirks/tuya/ts000f_switch.py

```diff
@@ -0,0 +1,91 @@
+"""tuya TS000F xkap8wtb On/Off In-line Switches with metering support."""
+
+from zigpy.profiles import zha
+from zigpy.zcl.clusters.general import (
+    Basic,
+    GreenPowerProxy,
+    Groups,
+    Identify,
+    OnOff,
+    Ota,
+    Scenes,
+    Time,
+)
+from zigpy.zcl.clusters.lightlink import LightLink
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODEL,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.tuya import (
+    TuyaZBE000Cluster,
+    TuyaZBElectricalMeasurement,
+    TuyaZBMeteringCluster,
+    TuyaZBOnOffAttributeCluster,
+)
+from zhaquirks.tuya.mcu import EnchantedDevice
+
+
+class Tuya_1G_Wall_Switch_Metering(EnchantedDevice):
+    """Tuya 1 gang wall switch with metering and restore power state support."""
+
+    signature = {
+        MODEL: "TS000F",
+        ENDPOINTS: {
+            # SizePrefixedSimpleDescriptor(endpoint=242, profile=41440, device_type=97, device_version=0, input_clusters=[], output_clusters=[33])
+            # input_clusters=["0x0000","0x0003","0x0004","0x0005","0x0006", "0x000a", "0x0702","0x0b04","0x1000","0x1888","0xe000"]
+            # output_clusters=["0x0019"]>
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_LIGHT,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    Identify.cluster_id,
+                    Groups.cluster_id,
+                    Scenes.cluster_id,
+                    OnOff.cluster_id,
+                    Time.cluster_id,
+                    TuyaZBMeteringCluster.cluster_id,
+                    TuyaZBElectricalMeasurement.cluster_id,
+                    LightLink.cluster_id,
+                    0x1888,
+                    TuyaZBE000Cluster.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [Ota.cluster_id],
+            },
+            # <SimpleDescriptor endpoint=242 profile=41440 device_type=97
+            # input_clusters=[]
+            # output_clusters=["0x0021"]>
+            242: {
```

Comment:
Its no reason deleting the GP EP its better to have it then its possible see its one GP router from the device card.

Reply:
@MattWestb can you propose a change here please?.. I'm sorry it's very hard to interpret "no reason deleting GP EP", maybe it's something obvious but I'm missing it completely... let me learn from you?

Reply:
Im sorry im not maintainer but @TheJulianJES can if hi have time doing that.

Reply:
@TheJulianJES apologies for interrupting you - I'm really not following the suggestion here, can you please invest a tiny bit of your precious time and share knowledge with @erkr and myself: what exactly is @MattWestb proposing?..

Reply:
Only adding the second endpoint in the replacement like normal Zigbee devices is having like this:
https://github.com/zigpy/zha-device-handlers/blob/74b9f043a996da6baca722d1e191d842a31ae198/zhaquirks/ikea/tradfriplug.py#L156-L166

I have taking some lines from the first ep with so you see how its being done.

Reply:
Thank you for pointing this snippet! Now I fully understand... I wasn't even aware about Zigbee Green Power (ZGP) before, just read up about the routing and sinking the metrics, now I understand why it's important to pass it on 🙏  Addressed this, resolving this review.